tests/NextButton.test.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { NextButton } from '../src/NextButton';
+
+const mockOnClick = jest.fn();
+
+const defaultProps = {
+    onClick: mockOnClick,
+};
+
+const renderWithProps = (props = {}) => render(<NextButton { ...defaultProps } { ...props }>foo</NextButton>);
+
+describe('NextButton Component', () => {
+    it('should render correctly', () => {
+        const { asFragment } = renderWithProps();
+
+        expect(asFragment()).toMatchSnapshot();
```

Comment:
Since we are using RTL here, I would test it differently. 

Instead of `toMatchSnapshot()` I would test it like this:

```js
const { getByText } = renderWithProps();
const button = getByText('foo');

expect(button.tagName).toMatch(/button/i);
expect(button).toHaveClass('next-button_button');
```

The advantage of this approach is that you don't need to update the snapshots of multiple tests if you make a functional change on this component.

Either way, it's just an opinion and changing this it's totally up to you.

Reply:
As we won't be running these tests many times I don't mind having snapshots, but I agree with you and don't mind changing this.

Although I'd argue that instead of matching by tagName we can:

```js
const { getByText, getByRole } = renderWithProps();
const button = getByRole('button');

expect(button).toHaveClass('next-button_button');
expect(getByText('foo')).toBeInTheDocument();
```